libazureinit/src/media.rs

```diff
@@ -71,6 +71,22 @@ fn default_preprov_type() -> String {
 pub const PATH_MOUNT_DEVICE: &str = "/dev/sr0";
 pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 
+const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
+
+// Get a mounted device with any filesystem for CDROM
+pub fn get_mount_device() -> Result<Vec<String>, Error> {
+    let mut list_devices: Vec<String> = Vec::new();
+
+    while let Some(device) = block_utils::get_mounted_devices()?
+        .iter()
+        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
+    {
+        list_devices.push(device.name.clone());
```

Comment:
```suggestion
        .into_iter()
        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
    {
        list_devices.push(device.name);
```
You can consume the returned list rather than iterating over references by using `into_iter()`. Then you won't need to clone the `device.name` since you own the struct and never reference it again.

Reply:
Done, thanks.